org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest2.java

```diff
@@ -0,0 +1,652 @@
+			this.mainAttributes.put(BUNDLE_REQUIREDEXECUTIONENVIRONMENT, bree);
+			this.modified = true;
+			this.bree = bree;
+		}
+	}
+
+	/**
+	 * @return the value of the main attribute
+	 * "Bundle-RequiredExecutionEnvironment".
+	 */
+	public String getBREE() {
+		return bree;
+	}
+
+	/**
+	 * Set the main attribute "Bundle-Activator" to the given value.
+	 * 
+	 * @param bundleActivator The new value
+	 */
+	public void setBundleActivator(String bundleActivator) {
+		String old = mainAttributes.get(BUNDLE_ACTIVATOR);
+		if (!bundleActivator.equals(old)) {
+			this.mainAttributes.put(BUNDLE_ACTIVATOR, bundleActivator);
+			this.modified = true;
+			this.bundleActivator = bundleActivator;
+		}
+	}
+
+	/**
+	 * @return the value of the main attribute "Bundle-Activator".
+	 */
+	public String getBundleActivator() {
+		return bundleActivator;
+	}
+
+	/**
+	 * Set the line delimiter to a specific value. Is only used for writing, NOT
+	 * for reading!
+	 * 
+	 * @param lineDelimeter typically either "\n" or "\r\n".
+	 */
+	public void setLineDelimiter(String lineDelimeter) {
+		newline = lineDelimeter;
+	}
+
+	/**
+	 * Write the contents to the manifest to the given stream.
+	 * 
+	 * @param stream the stream to write the output to.
+	 */
+	public void write(OutputStream stream) throws IOException {
+		write(new BufferedWriter(new OutputStreamWriter(stream)));
+	}
+
+	private void write(BufferedWriter writer) throws IOException {
+		writeHeader(writer);
+		writeEntries(writer);
+		writer.flush();
+	}
+
+	private void writeHeader(BufferedWriter writer) throws IOException {
+		if (version == null)
+			return;
+		String manifestVersion = mainAttributes.get(MANIFEST_VERSION);
+		if (manifestVersion != null)
+			writer.append(MANIFEST_VERSION).append(": ").append(manifestVersion).append(newline);
+		String signatureVersion = mainAttributes.get(SIGNATURE_VERSION);
+		if (signatureVersion != null)
+			writer.append(SIGNATURE_VERSION).append(": ").append(signatureVersion).append(newline);
+		for (Entry<String, String> entry : mainAttributes.entrySet()) {
+			String key = entry.getKey();
+			if (key.equals(MANIFEST_VERSION) || key.equals(SIGNATURE_VERSION))
+				continue;
+			String value = entry.getValue();
+			writer.append(make512Safe(new StringBuffer(key + ": " + separateCommas(value)), newline));
+		}
+	}
+
+	private void writeEntries(BufferedWriter writer) throws IOException {
+		for (Entry<String, Attributes> entry : entries.entrySet()) {
+			writer.write(newline);
+			writer.append("Name: ").append(entry.getKey()).append(newline);
+			for (Entry<String, String> child : entry.getValue().entrySet()) {
+				String key = child.getKey();
+				String value = child.getValue();
+				if (value.isEmpty())
+					continue;
+				writer.append(make512Safe(new StringBuffer(key + ": " + separateCommas(value)), newline));
+			}
+		}
+	}
+
+	private String separateCommas(String value) {
+		if (!value.contains(","))
+			return value;
+		String result = "";
```

Comment:
Not necessarily import in this context, but in general: result should be of type StringBuilder since we append in a loop.